tests/snappi_tests/multidut/pfcwd/files/pfcwd_multidut_burst_storm_helper.py

```diff
@@ -0,0 +1,331 @@
+        pause_pkt.pause_class_7.value = pause_time[7]
+
+        pause_flow_start_time = id * (pause_flow_dur_sec + pause_flow_gap_sec)
+
+        pause_flow.rate.pps = pause_pps
+        pause_flow.size.fixed = 64
+        pause_flow.duration.fixed_packets.packets = int(pause_pkt_cnt)
+        pause_flow.duration.fixed_packets.delay.nanoseconds = int(
+            sec_to_nanosec(pause_flow_start_time))
+
+        pause_flow.metrics.enable = True
+        pause_flow.metrics.loss = True
+
+
+def __run_traffic(api, config, all_flow_names, exp_dur_sec):
+    """
+    Run traffic and dump per-flow statistics
+
+    Args:
+        api (obj): SNAPPI session
+        config (obj): experiment config (testbed config + flow config)
+        all_flow_names (list): list of names of all the flows
+        exp_dur_sec (float): experiment duration in second
+
+    Returns:
+        per-flow statistics (list)
+    """
+    api.set_config(config)
+
+    logger.info('Wait for Arp to Resolve ...')
+    wait_for_arp(api, max_attempts=10, poll_interval_sec=2)
+
+    logger.info('Starting transmit on all flows ...')
+    ts = api.transmit_state()
+    ts.state = ts.START
+    api.set_transmit_state(ts)
+
+    time.sleep(exp_dur_sec)
+
+    attempts = 0
+    max_attempts = 20
+
+    while attempts < max_attempts:
+        request = api.metrics_request()
+        request.flow.flow_names = all_flow_names
+        rows = api.get_metrics(request).flow_metrics
+        """ If all the flows have stopped """
+        transmit_states = [row.transmit for row in rows]
+        if len(rows) == len(all_flow_names) and\
+           list(set(transmit_states)) == ['stopped']:
+            time.sleep(SNAPPI_POLL_DELAY_SEC)
+            break
+        else:
+            time.sleep(1)
+            attempts += 1
+
+    pytest_assert(attempts < max_attempts,
+                  "Flows do not stop in {} seconds".format(max_attempts))
+
+    """ Dump per-flow statistics """
+    request = api.metrics_request()
+    request.flow.flow_names = all_flow_names
+    rows = api.get_metrics(request).flow_metrics
+
+    logger.info('Stop transmit on all flows ...')
+    ts = api.transmit_state()
+    ts.state = ts.STOP
+    api.set_transmit_state(ts)
+
+    return rows
+
+
+def __verify_results(rows, data_flow_prefix, pause_flow_prefix):
+    """
+    Verify if we get expected experiment results
+
+    Args:
+        rows (list): per-flow statistics
+        data_flow_prefix (str): prefix of names of data flows
+        pause_flow_prefix (str): prefix of names of PFC pause storms
+
+    Returns:
+        N/A
+    """
+    for row in rows:
+        flow_name = row.name
+        tx_frames = row.frames_tx
+        rx_frames = row.frames_rx
+        logger.info('Flow Name : {} , Tx Frames : {}, Rx Frames : {}'.format(flow_name, tx_frames, rx_frames))
+
+        if data_flow_prefix in flow_name:
+            """ Data flow """
+            pytest_assert(tx_frames > 0 and rx_frames == tx_frames,
```

Comment:
Are we sending lossless traffic during this test ? If so, we are supposed to drop it during PFC storm.

Reply:
@vmittal-msft Multiple pause storms will be created (15) and each of them will be running for a duration of 0.1s and the gap between these pause storms will be sum of ("polling interval" + "detection-time" + "restoration-time" + 0.1). As the pause storms are not continuous, pfcwd should not hit and should not affect lossless traffic.

Reply:
@kamalsahu0001 Just to make sure and make debugging later easier, can you add a `logger.info()` with the pfcwd status of the switch such as the output of `show pfcwd status` so that we can confirm this to be true if this case fails.

Reply:
@developfast Requested changes done.